deep_qa/data/instances/sentence_selection_instance.py

```diff
@@ -0,0 +1,161 @@
+    def read_from_line(cls, line: str, default_label: bool=None):
+        """
+        Reads a SentenceSelectionInstance object from a line.
+        The format has one of two options:
+
+        (1) [example index][tab][question][tab][list_of_sentences][tab][label]
+        (2) [question][tab][list_of_sentences][tab][label]
+
+        The ``list_of_sentences`` column is assumed formatted as: ``[sentence]###[sentence]###[sentence]...``
+        That is, we split on three hashes (``"###"``).
+
+        ``default_label`` is ignored, but we keep the argument to match the interface.
+        """
+        fields = line.split("\t")
+
+        if len(fields) == 4:
+            index_string, question, sentences, label_string = fields
+            index = int(index_string)
+        elif len(fields) == 3:
+            question, sentences, label_string = fields
+            index = None
+        else:
+            raise RuntimeError("Unrecognized line format: " + line)
+        # get the answer options
+        sentences_split = sentences.split("###")
+        label = int(label_string)
+
+        return cls(question, sentences_split, label, index)
+
+
+class IndexedSentenceSelectionInstance(IndexedInstance):
+    """
+    This is an indexed instance that is used for the answer sentence selection task,
+    where we have a question, a list of candidate sentences, and an
+    int label that indicates the index of the correct sentence in the list of
+    candidate sentences.
+    """
+    def __init__(self,
+                 question_indices: List[int],
+                 sentences_indices: List[List[int]],
+                 label: int,
+                 index: int=None):
+        super(IndexedSentenceSelectionInstance, self).__init__(label, index)
+        self.question_indices = question_indices
+        self.sentences_indices = sentences_indices
+
+    @classmethod
+    @overrides
+    def empty_instance(cls):
+        return IndexedSentenceSelectionInstance([], [], label=None, index=None)
+
+    @overrides
+    def get_lengths(self) -> Dict[str, int]:
+        """
+        We need to pad at least the question length, the passage length, and the
+        word length across all the questions and passages. Subclasses that
+        add more arguments should also override this method to enable padding on said
+        arguments.
+        """
+        question_lengths = self._get_word_sequence_lengths(self.question_indices)
+        sentences_lengths = [self._get_word_sequence_lengths(sentence_indices)
+                             for sentence_indices in self.sentences_indices]
+        lengths = {}
+
+        # the number of sentences
+        lengths['num_sentences'] = len(self.sentences_indices)
+
+        # the number of words in the question
+        lengths['num_question_words'] = question_lengths['num_sentence_words']
+
+        # the number of words in each passage
+        lengths['num_passage_words'] = max(sentence_lengths['num_sentence_words']
```

Comment:
ah, most of the other `get_lengths` functions have standardized to use `num_passage_words` so i was concerned that changing it would break things that interface with the instance. `sentence` should be fine though.